### PR TITLE
camel-spring-rabbitmq - new component #2128

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/2128

There were some auto-generated changes (after `mvn process-resources -Pformat`). I put them into the second commit.

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: extensions/spring-rabbitmq/runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-spring-rabbitmq</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-support-commons-logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-reflect</artifactId>
```
Comment (jamesnetherton) [inline extensions/spring-rabbitmq/runtime/pom.xml:67]: `kotlin-reflect` may actually not be required if you add a dependency on `camel-quarkus-support-spring`. It also helps to ensure our Spring backed extensions play well with the Quarkus Spring stuff.

**Review by oscerd — APPROVED** [review]

File: integration-tests/spring-rabbitmq/pom.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
```
Comment (jamesnetherton) [inline integration-tests/spring-rabbitmq/pom.xml:102]: Duplicate comment - FYI @ppalaga, I think there's a bug in the cq-maven-plugin that causes this.

**Review by davsclaus — APPROVED** [review]

File: extensions/aws-ec2/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 3]
```diff
@@ -31,3 +31,4 @@ metadata:
   - "integration"
   status:
   - "stable"
+  - "deprecated"
```
Comment (ppalaga) [inline extensions/aws-ec2/runtime/src/main/resources/META-INF/quarkus-extension.yaml:34]: These deprecation changes would deserve a separate commit. They belong to the Camel 3.8 upgrade.

Comment (JiriOndrusek) [reply extensions/aws-ec2/runtime/src/main/resources/META-INF/quarkus-extension.yaml:34]: @ppalaga  I've put deprecation changed into separated commit in this PR, should I create another PR?

Comment (ppalaga) [reply extensions/aws-ec2/runtime/src/main/resources/META-INF/quarkus-extension.yaml:34]: This commit is OK

File: extensions/spring-rabbitmq/runtime/pom.xml (hunk 1 above)
Comment (JiriOndrusek) [reply extensions/spring-rabbitmq/runtime/pom.xml:67]: ok, thanks for the information, I'll try it  - it could be much better solution.

Comment (JiriOndrusek) [reply extensions/spring-rabbitmq/runtime/pom.xml]: you are right, there is a substitution in `camel-quarkus-support-spring`, which makes kotlin not required. I've replaced the dependencies.

File: integration-tests/spring-rabbitmq/pom.xml (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/pom.xml]: I've removed duplicity.

Comment (ppalaga) [reply integration-tests/spring-rabbitmq/pom.xml]: Good to know. It would be nice to keep an eye on this when running `mvn process-resources -Pformat` and give hints about situations when it happens.

**Review by ppalaga — APPROVED** [review]

File: integration-tests/spring-rabbitmq/pom.xml (hunk 2 above)
Comment (ppalaga) [reply integration-tests/spring-rabbitmq/pom.xml]: I think I was able to pinpoint it: it happens when the newly inserted dependeny is placed before  the exiting ones. In that case the plugin chooses a bd insert location.

Thanks for notifying me!
